ios/FluentUI/Vnext/Avatar/AvatarGroup.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+@objc public enum MSFAvatarGroupStyle: Int, CaseIterable {
+    case stack
+    case pile
+}
+//
+/// `MSFAvatarGroupState` defines properties in an AvatarGroup
+///
+/// `maxDisplayedAvatars`: Maximum number of avatars displayed
+///
+/// `overflowCount`: Number to be displayed in the overflow avatar
+///
+/// `style`: Type of AvatarGroup. Avatars in a Stack is layered on top of each other; Pile separates avatars
+///
+/// `createAvatar`: Create an Avatar using an AvatarStyle and AvatarSize
+///
+/// `getAvatarState`: Get the state of an Avatar using an index value
+///
+/// `deleteAvatar`: Delete an Avatar with a specified index value
+@objc public protocol MSFAvatarGroupState {
+    var maxDisplayedAvatars: Int { get set }
+    var overflowCount: Int { get set }
+    var style: MSFAvatarGroupStyle { get set }
+
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize)
+    func getAvatarState(_ atIndex: Int) -> MSFAvatarState
+    func deleteAvatar(atIndex: Int)
+}
+
+/// Properties that make up AvatarGroup content
+class MSFAvatarGroupStateImpl: NSObject, ObservableObject, MSFAvatarGroupState {
+    func createAvatar(style: MSFAvatarStyle, size: MSFAvatarSize) {
```

Comment:
Question about design for these. I would have assumed that these `State` objects were meant to be "dumb" property bags, without any logic. This isn't *much* logic, but it does break that assumption.

Reply:
This new design change was made to address this [comment](https://github.com/microsoft/fluentui-apple/pull/582#discussion_r645686438). This is so that the consumer does not have the responsibility to create the avatar state as this class will handle it instead. @rdeassis

Reply:
Makes sense. I agree with the earlier comments that clients shouldn't have to create views, but I'm not sure if this is the approach we should be taking. I'm actually running into similar questions regarding `PersonaCarousel` right now: what should the public interface be for creating these?

@rdeassis and I discussed very briefly offline yesterday, but we kinda settled on a pattern like the following:
1. Caller instantiates the collection (e.g. `AvatarGroup`)
2. Caller instantiates one or more data model structs/objects that represents the items to add (e.g. `Avatar`)
3. Caller fills out all properties on the local item data model(s)
4. Caller passes the now-fully-formed object(s) to the container to add it/them

Now the big sticking point: what is the object created in step 2? I don't believe we've fully come to an agreement on that point yet. Hopefully we all can discuss this further early next week.

Reply:
(Note: that list of steps should also include an alternative where the avatar data is passed in at `init` time.)

Just for reference, check out how the "legacy" AvatarGroupView is created and populated: https://github.com/microsoft/fluentui-apple/blob/9abc10617ed457679e63734b7f7f3372295502a6/ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarGroupViewDemoController.swift#L182

Reply:
I noticed that and had thought about creating some sort of convenience method, should I do that here as well?

Reply:
As discussed offline, let's hold off on anything too big at the moment, though I'm intrigued about this convenience method.